x-pack/plugin/security/src/main/java/org/elasticsearch/xpack/security/authz/AuthorizationService.java

```diff
@@ -502,34 +554,57 @@ private void authorizeAction(
                         requestInfo,
                         authzInfo,
                         projectMetadata.getIndicesLookup(),
-                        ActionListener.wrap(
-                            authorizedIndices -> resolvedIndicesListener.onResponse(
-                                indicesAndAliasesResolver.resolve(action, request, projectMetadata, authorizedIndices)
-                            ),
-                            e -> {
-                                if (e instanceof InvalidIndexNameException
-                                    || e instanceof InvalidSelectorException
-                                    || e instanceof UnsupportedSelectorException) {
-                                    logger.debug(
-                                        () -> Strings.format(
-                                            "failed [%s] action authorization for [%s] due to [%s] exception",
+                        ActionListener.wrap(authorizedIndices -> {
+                            if (request instanceof IndicesRequest.Replaceable replaceable && replaceable.resolveCrossProject()) {
+                                crossProjectSearchAuthzService.loadAuthorizedProjects(ActionListener.wrap(authorizedProjects -> {
```

Comment:
The reason this is here (rather than in IndicesAndAliasesResolver) is that it's an async call. It's an async call because we make a network call to the UIAM service to retrieve the list of authorized projects. 

We could refactor `indicesAndAliasesResolver.resolve` to work async but I'm leaning towards the split as it currently is (async resolve projects here, do the sync rewrite in `indicesAndAliasesResolver.resolve`).